components/match2/commons/template_match.lua

```diff
@@ -108,14 +108,13 @@ function TemplateMatch._recursiveSetBracketIndex(matches, id, headerchild, appli
 		return matches, applied
 	end
 	id = TemplateMatch._getTrueID(id)
+	---@cast id -nil
```

Comment:
Move it up one line
Update the `TemplateMatch._getTrueID` function

Reply:
an mobile atm but iirc the function is used on some other places in this file too
so need to check of we actually can adjust it so easily

Reply:
Can add annotations to getTrueId

```
---@param id string
---@return string
---@overload fun(): nil
```

Also fix the return by doing `return (...)`, else it's returning two values (as gsub returns that)